src/components/plots/rectanglePlot.ts

```diff
@@ -0,0 +1,110 @@
+///<reference path="../../reference.ts" />
+
+module Plottable {
+export module Plot {
+  export class Rectangle extends AbstractXYPlot<any, any> {
+    private _colorScale: Scale.AbstractScale<any, string>;
+
+    /**
+     * Constructs a RectanglePlot.
+     *
+     * A RectanglePlot consists of a bunch of rectangles. The user is required to
+     * project the top left corner of each rectangle (x1, y1) and the bottom right
+     * corner of each rectangle (x2, y2)
+     *
+     * @constructor
+     * @param {Scale.Ordinal} xScale The x scale to use.
+     * @param {Scale.Ordinal} yScale The y scale to use.
+     */
+    constructor(xScale: Scale.AbstractScale<any, any>, yScale: Scale.AbstractScale<any, any>,
+        colorScale: Scale.AbstractScale<any, string>) {
+      super(xScale, yScale);
+
+      // Category scales should not have padding by default
+      if (xScale instanceof Scale.Ordinal) {
+        (<Scale.Ordinal> xScale).innerPadding(0).outerPadding(0);
+      }
+      if (yScale instanceof Scale.Ordinal) {
+        (<Scale.Ordinal> yScale).innerPadding(0).outerPadding(0);
+      }
```

Comment:
Err we should be careful here since this works on gridplots but not barPlots or any of the other plots (unless we want them to have 0 padding)